### PR TITLE
map_tax: [FIX] signatures

### PR DESCRIPTION
**Pull request by blaggacao**
Towards #23333

As to `partner_shipping_id` vs `partner_id` I would suggest an internal review (iirc, I spotted an inconsistency between analytic and normal lines in the context of sale orders). Not 100% sure though, so I would leave that as a suggestion...

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Some call signatures have not been aligned with the taxes/product/partner signature
introduced by the tax cloud module and necesary by some localization modules.

### CODE REVIEW

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -1017,7 +1017,7 @@ def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, val
 
         taxes = product_id.supplier_taxes_id
         fpos = po.fiscal_position_id
-        taxes_id = fpos.map_tax(taxes) if fpos else taxes
+        taxes_id = fpos.map_tax(taxes, self.product_id, seller.name) if fpos else taxes
```
Comment (mart-e) [inline addons/purchase/models/purchase.py]: shouldn't it be the `product_id` given in parameter?

Comment (blaggacao) [reply addons/purchase/models/purchase.py]: True

**Comment by mart-e** [comment]
Merged at 56eead83568, thanks

**Comment by blaggacao** [comment]
@mart-e Thanks for your collaboration! I really appreciate it!

**Pull request closed without merging**